### ISSUE TITLE
Implement particle-based lighting

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot-proposals#7606]
### Describe the project you are working on

VFX and Godot

### Describe the problem or limitation you are having in your project

Particles can be emissive, but they can't have the same effect as an omnilight.

Example in Unity: https://docs.unity3d.com/Manual/PartSysLightsModule.html

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a way to make particle emit and affect the environment as if they were each a point light

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Particles are done in a compute step of the pipeline, I suppose it should be possible to displace a set amount of lights after the compute step. 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No - this is very deep in the render pipeline

### Is there a reason why this should be core and not an add-on in the asset library?

See above

**Comment by Calinou** [comment]
Note that light rendering is relatively expensive (especially if not using Forward+), so this will require adding a property to define the ratio of particles that emit light. For instance, if you set Amount to 50 and the light emission ratio to 0.2, only 10 particles can emit a light at most.

There also needs to be a way to the animate light color, energy and radius over time, which may complexify the particle setup a lot.

**Comment by Devouring-One** [comment]
Came here to bump this, I don't want to create my own particle system (but it seems that I have to...)

**Comment by QbieShay** [comment]
@Devouring-One I don't think this will be done any time soon. What's your usecase? Can you give more details?

**Comment by Stwend** [comment]
One usecase (at least for me) would be having cheap large scale "AO" for smoke puffs, by including a few large, slightly negative intensity lights in the particle system (not affecting the particles themselves) to take out some brightness in the environment around the smoke. That effect could also be used as a subtle "contrast boost" around light emitting particles.

**Comment by QbieShay** [comment]
@Stwend this should already be possible to do by using some subtractive blending and some spherical shader on spheres/cards, i think.

**Comment by Stwend** [comment]
@QbieShay Contrast boost, absolutely, but I don't think simulating negative point lights using cards or spheres works.

**Comment by Calinou** [comment]
Here's a proof of concept of faking lights with billboarded quads that use proximity fade (standard and reverse) and distance fade, using a custom shader:

### Additive lighting

https://github.com/user-attachments/assets/bbf3448b-581f-4adb-9b0e-b2e355d8ce6d

### Subtractive lighting

https://github.com/user-attachments/assets/3dd8b6d4-339a-4e56-928c-b603bb17757b

This does not use actual lights, so it can be faster, but the effect is less convincing, especially in dark areas. With CPUParticles, it's technically possible to parent actual lights to them with some C++ code changes, as seen in https://github.com/godotengine/godot/pull/61273.

This technique works in all renderers, including Compatibility.

Project files (licensed under CC0): [test_particle_lights.zip](https://github.com/user-attachments/files/24084624/test_particle_lights.zip)

**Comment by QbieShay** [comment]
@Calinou thank you for your contribution!!
